documentation/src/main/asciidoc/user_guide/clustered_lock.adoc

```diff
@@ -0,0 +1,200 @@
+== Clustered Lock
+
+Clustered lock is a new building block introduced in Infinispan 9.2
+It is a lock distributed and shared among all nodes in the Infinispan cluster and currently provides a way to execute
+code that will be synchronized between the nodes in a given cluster.
+
+=== Installation
+
+In order to start using the clustered locks, you needs to add the dependency in your Maven `pom.xml` file:
+
+.pom.xml
+[source,xml]
+----
+<dependency>
+   <groupId>org.infinispan</groupId>
+   <artifactId>infinispan-clustered-lock</artifactId>
+   <version>...</version> <!-- 9.2.0.Final or higher -->
+</dependency>
+----
+
+Currently no extra configuration needs to be provided.
+
+=== The ClusteredLockConfiguration.
+
+Currently there is a single type of `ClusteredLock` supported : non reentrant, NODE ownership lock.
+
+==== Ownership
+
+* `NODE`
+When a `ClusteredLock` is defined, this lock can be used from all the nodes in the infinispan cluster.
+When the ownership is NODE type, this means that the owner of the lock is the infinispan node that acquired the lock
+at a given time.
+This means that each time we get a `ClusteredLock` instance with the `ClusteredCacheManager`, this instance will be the
+same instance for each infinispan node.
+This lock can be used to synchronize code between infinispan nodes. The advantage of this lock is that any thread in the
+node can release the lock at a given time.
+
+* `INSTANCE` - not yet supported
+
+When a `ClusteredLock` is defined, this lock can be used from all the nodes in the infinispan cluster.
+When the ownership is INSTANCE type, this means that the owner of the lock is the actual instance we acquired when
+`ClusteredLockManager.get("lockName")` is called.
+
+This means that each time we get a `ClusteredLock` instance with the `ClusteredCacheManager`, this instance will be
+a new instance.
+This lock can be used to synchronize code between infinispan nodes and inside each infinispan node.
+The advantage of this lock is that only the instance that called 'lock' can release the lock.
+
+
+==== Reentrancy
+
+When a `ClusteredLock` is configured reentrant, this means the owner of the lock can reacquire the lock as many consecutive
+times as it wants while holding the lock.
+
+Currently, only non reentrant locks are supported. This means that when two consecutive `lock` calls are sent for the same
+owner, the first call will acquire the lock if it's available, and the second call will block.
+
+
+=== The ClusteredLockManager interface.
+
+The `ClusteredLockManager` interface, *marked as experimental*, is the entry point to define, retrieve and remove a lock.
+It automatically listen to the creation of `EmbeddedCacheManager` and proceeds with the registration  of an
+instance of it per `EmbeddedCacheManager`.
+It starts the caches needed to store the lock state.
```

Comment:
"It starts the internal caches needed..."

Reply:
ok